website/docs/getting_started.html.markdown

```diff
@@ -0,0 +1,224 @@
+    }
+  }
+}
+```
+
+~> Note: Don't use `terraform apply` quite yet! You still need to add GCP
+credentials below. If you want to try out provisioning your VM instance before
+continuing, follow the instructions in the "Adding credentials" section below.
+
+## Connecting to a VPC network
+
+Like this VM instance, nearly every GCP resource will have a `name` field. They
+are used as a short way to identify resources, and their name in the Cloud
+Console will be the one defined in the `name` field.
+
+When linking resources in a Terraform config though, you'll primarily want to
+use a different field, the `self_link` of a resource. Like `name`, nearly every
+resource has a `self_link`. They look like:
+
+```
+{{API base url}}/projects/{{your project}}/{{location type}}/{{location}}/{{resouce type}}/{{name}}
+```
+
+For a real example, let's look at what the instance `self_link` will be assuming
+your project is named `foo`:
+
+```
+https://www.googleapis.com/compute/v1/projects/foo/zones/us-central1-c/instances/terraform-instance
+```
+
+A resource's `self_link` is a unique reference to that resource. When
+connecting two resources in Terraform, you can use Terraform interpolation to
+avoid typing all that out! Let's use a `google_compute_network` to demonstrate.
+
+Add this block to your config:
+
+```hcl
+resource "google_compute_network" "vpc_network" {
+  name                    = "terraform-network"
+  auto_create_subnetworks = "true"
+}
+```
+
+This will create [VPC network resource](/docs/providers/google/r/compute_network.html)
+with a subnetwork (due to auto_create_subnetworks) in each region. Then, change
+the network of the `google_compute_instance`.
+
+```diff
+network_interface {
+-  # A default network is created for all GCP projects
+-  network = "default"
++  network = "${google_compute_network.vpc_network.self_link}"
+  access_config = {
+```
+
+This means that when we create the VM instance, it will use
+`"terraform-network"` instead of the default VPC network for the project. If you
+run `terraform plan`, you should see that `"terraform-instance"` depends on
+`"terraform-network"`.
+
+You're ready to run `terraform apply`! Except for one last thing...
+
+~> You haven't added GCP API credentials yet.
+
+## Adding credentials
+In order to make requests against the GCP API, you need to authenticate to prove
+that it's you making the request. The preferred method of provisioning resources
+with Terraform is to use a [GCP service account](https://cloud.google.com/docs/authentication/getting-started),
+a "robot account" that can be granted a limited set of IAM permissions.
+
+From [the service account key page in the Cloud Console](https://pantheon.corp.google.com/apis/credentials/serviceaccountkey)
+choose an existing account, or create a new one. Then, download the JSON key
+file. Name it something you can remember, and store it somewhere secure on your
+machine.
+
+You supply the key to Terraform using the environment variable
```

Comment:
This is for default credentials, which I'm a bit nervous about telling people to do since I'd assume that it would get overridden if you ended up doing a `gcloud auth application-default login` for some reason. I'd rather tell people to use the explicit env variables that we provide (`GOOGLE_CREDENTIALS` being the preferred) and then to use `export GOOGLE_CREDENTIALS=$(cat {{path}})`

Reply:
Oh yup I totally thought I had used that env var myself but I was wrong I used `GOOGLE_CLOUD_KEYFILE_JSON`. Updated the credentials definition to say paths also work.